cypress/e2e/signIn.cy.js

```diff
@@ -10,29 +10,37 @@ const homePage = new HomePageObject();
 describe('Sign In page', () => {
   let user;
 
-  before(() => {
+  beforeEach(() => {
     cy.task('db:clear');
     cy.task('generateUser').then(generateUser => {
       user = generateUser;
     });
+    signInPage.visit();
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.emailField
-      .type(user.email);
-    signInPage.passwordField
-      .type(user.password);
-    signInPage.signInBtn
-      .click();
-
+    cy.registerNewUser().then(user => {
+      cy.getByDataCy('email-sign-in').type(user.email);
+      cy.getByDataCy('password-sign-in').type(user.password);
+      cy.getByDataCy('sign-in-btn').click();
+      cy.getByDataCy('username-link', user.username).should('exist');
     homePage.usernameLink
       .should('contain', user.username);
   });
+});
 
-  it('should not provide an ability to log in with wrong credentials', () => {
+  it('should not provide an ability to log in with wrong email', () => {
+    signInPage.emailField.type(`wrong${user.email}`);
```

Comment:
Instead of hardcoding error messages in tests like 'Invalid user credentials.', you can store them in a separate module and import when needed. This strategy will help you maintain tests easier as you will not need to change tests when error messages are updated, just the module.